pkg/controller/tidbcluster/pod_control.go

```diff
@@ -179,6 +179,13 @@ func (c *PodController) sync(key string) (reconcile.Result, error) {
 		return reconcile.Result{}, nil
 	}
 
+	tc, err := c.deps.TiDBClusterLister.TidbClusters(ns).Get(tcName)
```

Comment:
IMHO, this will still print the error logs, the root cause of #4447 here is that we are watching all pods rather than pods that are filtered out by the lable selector.

Reply:
Agree, but unfortunately, pods don't contain labels that are set by `controllerManager.selector`.

Reply:
Maybe change `deps.KubeInformerFactory` to `deps.LabelFilterKubeInformerFactory` will work
https://github.com/pingcap/tidb-operator/blob/c09574aa38346ab52ead0498bd6cf282a4f58abc/pkg/controller/tidbcluster/pod_control.go#L67
or change the podLister in deps directlly?
https://github.com/pingcap/tidb-operator/blob/c09574aa38346ab52ead0498bd6cf282a4f58abc/pkg/controller/dependences.go#L330

Reply:
I see what you mean, that doesn't help. : (

Reply:
If don't need to watch Pods, we can list TC firstly and find their Pods. WDYT? @july2993

Reply:
The annotation is added on the pod to trigger a restart, to be responsive we need to watch pod.
how about

Reply:
This can avoid the issue of #4447, but we should just return nil if and only the error is NotFound because the tc may already be deleted when we handle the pod.
And seems we should ignore the pod that belong to tc with labels not match our `controllerManager.selector`

Reply:
`c.deps.TiDBClusterLister` have filter the tc not match `controllerManager.selector`

Reply:
> This can avoid the issue of #4447, but we should just return nil if and only the error is NotFound because the tc may already be deleted when we handle the pod.

It makes sense to me, we do nothing if `deps.TiDBClusterLister` returns NotFoundError, which means either tc is deleted or tc is not match `selector`.

Reply:
then only ignore not found error should be ok here

Reply:
> then only ignore not found error should be ok here

Done